WordPress/WordPressShare/BlogPickerViewController.swift

```diff
@@ -0,0 +1,135 @@
+import Foundation
+import UIKit
+import WordPressShared
+import WordPressComKit
+
+
+class BlogPickerViewController : UITableViewController
+{
+    // MARK: - View Lifecycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupView()
+        setupTableView()
+        setupNoResultsView()
+        loadSites()
+    }
+    
+    
+    // MARK: - UITableView Methods
+    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sites?.count ?? 0
+    }
+    
+    override func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
+        return rowHeight
+    }
+    
+    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        var cell = tableView.dequeueReusableCellWithIdentifier(reuseIdentifier)
+        if cell == nil {
+            cell = WPTableViewCell(style: .Subtitle, reuseIdentifier: reuseIdentifier)
+            WPStyleGuide.Share.configureBlogTableViewCell(cell!)
+        }
+        
+        let site = sites?[indexPath.row]
+        configureCell(cell!, site: site!)
+        
+        return cell!
+    }
+    
+    override func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath) {
+        let site = sites![indexPath.row]
+        onChange?(siteId: site.ID, description: site.name?.characters.count > 0 ? site.name : site.URL.absoluteString.hostname())
+        navigationController?.popViewControllerAnimated(true)
+    }
+    
+    
+    // MARK: - Setup Helpers
+    private func setupView() {
+        title = NSLocalizedString("Site Picker", comment: "Title for the Site Picker")
+        preferredContentSize = UIScreen.mainScreen().bounds.size
+    }
+    
+    private func setupTableView() {
+        // Blur!
+        let blurEffect = UIBlurEffect(style: .Light)
+        tableView.backgroundColor = UIColor.clearColor()
+        tableView.backgroundView = UIVisualEffectView(effect: blurEffect)
+        tableView.separatorEffect = UIVibrancyEffect(forBlurEffect: blurEffect)
+        
+        // Fix: Hide the cellSeparators, when the table is empty
+        tableView.tableFooterView = UIView()
+    }
+    
+    private func setupNoResultsView() {
+        noResultsView = WPNoResultsView()
+        tableView.addSubview(noResultsView)
+    }
+    
+    
+    // MARK: - Private Helpers
+    private func loadSites() {
+        let authDetails = ShareExtensionService.retrieveShareExtensionConfiguration()
```

Comment:
I'd even argue that since you can't present this controller without a configuration, pass that when initializing. Or initialize with the service. But then it looks like the OAuth token is set on `RequestRouter` which looks like a singleton thing, so it could be already set by the caller.